File: src/omnia/data/structure/mutable/MutableGraph.java
package omnia.data.structure.mutable;

import omnia.data.structure.Graph;

public interface MutableGraph<E> extends Graph<E> {

  void addNode(E item);

  void replaceNode(E original, E replacement);

  default boolean removeNode(E item) {
    return removeUnknownTypedNode(item);
  }

  boolean removeUnknownTypedNode(Object item);
}

File: src/omnia/data/structure/mutable/MutableGraph.kt
package omnia.data.structure.mutable

import omnia.data.structure.Graph

interface MutableGraph<E> : Graph<E> {
  fun addNode(item: E)
  fun replaceNode(original: E, replacement: E)
  fun removeNode(item: E): Boolean {
    return removeUnknownTypedNode(item)
  }

  fun removeUnknownTypedNode(item: Any?): Boolean
}